src/common/utils/usePagination.ts

```diff
@@ -1,21 +1,30 @@
 import { useHistory } from 'react-router-dom';
 import { useCallback } from 'react';
+import { History } from 'history';
 
 const PAGE_SIZE = 10;
 
 export const usePagination = (pageSize = PAGE_SIZE) => {
   const history = useHistory();
-
-  const urlSearchParams = new URLSearchParams(history.location.search);
-  const pageParam = Number(urlSearchParams.get('page') ?? 1);
+  const pageParam = getCurrentPageParam(history);
   const pageIndex = !Number.isNaN(pageParam) ? pageParam - 1 : 0;
   const cursorIndex = pageIndex * pageSize;
   const cursor = cursorIndex > 0 ? btoa(`arrayconnection:${cursorIndex - 1}`) : undefined;
 
   const getPageCount = (connectionsCount: number | null | undefined) =>
     connectionsCount ? Math.ceil(connectionsCount / pageSize) : 1;
 
-  const goToPage = useCallback((pageIndex: number) => history.push({ search: `?page=${pageIndex + 1}` }), [history]);
+  const goToPage = useCallback(
+    (pageIndex: number) => {
+      const newPageNumber = pageIndex + 1;
+      const currentPageNumber = getCurrentPageParam(history);
```

Comment:
Why do we need to extract the logic to `getCurrentPageParam`? why don't we just access `pageParam` here?

Reply:
The `pageParam` that is visible as closure is not updated after the component is initialized, so we need to get it from URL.

Reply:
Makes sense.